lib/ClangImporter/ImporterImpl.h

```diff
@@ -528,6 +528,10 @@ class LLVM_LIBRARY_VISIBILITY ClangImporter::Implementation
   /// used when parsing the attribute text.
   llvm::SmallDenseMap<ModuleDecl *, SourceFile *> ClangSwiftAttrSourceFiles;
 
+  /// If set, the Swift DeclContext from which the current
+  /// import request was triggered.
+  llvm::Optional<const swift::DeclContext *> ImportingContext;
```

Comment:
For the purposes of this patch, all that matters is that the provided `DeclContext` be within the parent `SourceFile` `DeclContext`. However, other use cases might require the `DeclContext` that is the direct parent of the usage? I don't know how easy it will be to pass that information to the lookup in all cases, so I think maybe its best to rename to SourceFileContext, or even just store a `SourceFile`.